packages/okta-react/src/AuthService.js

```diff
@@ -196,12 +190,24 @@ class AuthService {
   }
 
   async login(fromUri, additionalParams) {
+    if(this._pending.handleLogin) { 
```

Comment:
nice. thanks!

Reply:
the closer the easier to manage, hah

Reply:
If login should only be run once, we probably can move the check to auth-js v5.0.

Reply:
The underlying `token.getWithRedirect` will be placed in the PromiseQueue, but there is additional logic here. One possible error would be calling `login(myUri)` and then calling `login` again with no parameters. This would overwrite the saved URI and affect the final redirect value.

Another option instead of silently ignoring a concurrent `login` attempt is to actually error out. Maybe we could do this in the next release? What I'd like is a way to compare authState with previous authState so we only trigger login if we are transitioning `isAuthenticated` from pending -> false or true -> false. A transition of false -> false should not start a login flow.

Reply:
> if we are transitioning isAuthenticated from pending -> false or true -> false

Interesting.  I think that level of check is better done in auth-js, but either way it does look like a nice approach to guide developers away from pitfalls that can otherwise _mostly_ work.

Reply:
`login` might not only be triggered by `isAuthenticated` state change,  users may also call it from callbacks which is out of our control. I think it still makes sense to protect it generally from concurrency cases.